tests/loader/test_yfcc_imb_loader.py

```diff
@@ -0,0 +1,126 @@
+"""Unit tests for the YFCC IMB loader.
+
+Authors: Ayush Baid
+"""
+import unittest
+from pathlib import Path
+
+import numpy as np
+from gtsam import Pose3, Cal3Bundler
+
+import gtsfm.utils.io as io_utils
+from gtsfm.loader.yfcc_imb_loader import YfccImbLoader
+
+DATA_ROOT_PATH = Path(__file__).resolve().parent.parent / "data"
+FOLDER_PATH = DATA_ROOT_PATH / "imb_reichstag"
+
+
+class TestYfccIMbLoader(unittest.TestCase):
+    """Unit test for the YFCC IMB Loader."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.loader = YfccImbLoader(str(FOLDER_PATH))
+
+    def test_len(self):
+        """Test the number of entries in the loader."""
+
+        expected = 10
+
+        self.assertEqual(len(self.loader), expected)
+
+    def test_get_image_valid_index(self):
+        """Tests that get_image works for all valid indices."""
+
+        for idx in range(len(self.loader)):
+            self.assertIsNotNone(self.loader.get_image(idx))
+
+    def test_get_image_invalid_index(self):
+        """Test that get_image raises an exception on an invalid index."""
+
+        # negative index
+        with self.assertRaises(IndexError):
+            self.loader.get_image(-1)
+        # len() as index
+        with self.assertRaises(IndexError):
+            self.loader.get_image(10)
+        # index > len()
+        with self.assertRaises(IndexError):
+            self.loader.get_image(100)
+
+    def test_image_contents(self):
+        """
+        Test the actual image which is being fetched by the loader at an index.
+
+        This test's primary purpose is to check if the ordering of filename is being respected by the loader
+        """
+
+        index_to_test = 5
+        file_path = FOLDER_PATH / "images" / "05866831_3427466899.jpg"
+
+        loader_image = self.loader.get_image(index_to_test)
+
+        expected_image = io_utils.load_image(file_path)
+
+        np.testing.assert_allclose(
+            expected_image.value_array, loader_image.value_array
+        )
```

Comment:
nit and more of personal preference: Isn't that too many empty lines between code?